### PR TITLE
Dynamic of thumbail

### LINKED ISSUE #177: Dynamic Profile
Aim at making the recipe thumbnail display under the profile dynamic and not hardcoded.

Require to link a created recipe to the profile who created them
Require to link a create comment to the profile who created them
Require to link a create comment to the recipe they were commented on.

More detail are on the PR description!

### PR DESCRIPTION
**Pull request by fdejean**
This PR aim at making the recipe thumbnail display under the profile dynamic and not hardcoded.
Hence,  when a User create a recipe, it will now be display in his profile !!

- [x] Transaction function Link created recipe to the profile who created them
- [x] Transaction function Link create comment to the profile who created them
- [x] Transaction function Link create comment to the recipe they were commented on.
- [x] Update UI to display recipe fetched from the database
- [x] Update ViewModel to fetch the recipe created by a certain profile

Those implied the following modification :

- [x] new field on the recipe data model : commentId list
- [x] Adapt all the test that used a recipe due to the update of the model
- [x] Modified created recipe to wait until the recipe is uploaded to the database and refetched the profile to display the newly created recipe

### CODE REVIEW

**Comment by fdejean** [comment]
The feature work, a minor bug is that we did not resize image upon input/upload, hence create recipe thumbnail format is not consistent, however the UI is not broken.
The coverage is missing 30lines that are hard to test due to the nature of the interaction :
Need to mock the Profile, Comment and Recipe database and every interaction call, in the same place and I could not do it simply with our test infrastructure. 
Hence it should be a new PR as it will take a few hours to correctly set them up...

File: app/src/main/java/com/android/feedme/ui/profile/RecipeInputScreen.kt [hunk 1]
```diff
@@ -107,9 +107,16 @@ fun RecipeInputScreen(
...
-                navigationActions.navigateTo(Route.PROFILE)
+
+                recipeViewModel.setRecipe(recipeViewModel.recipe.value!!) {
+                  profileViewModel.addRecipe(recipeViewModel.recipe.value!!)
+                  // unsure the UI is up to date
+
+                }
               }
+              navigationActions.navigateTo(Route.PROFILE)
+              // TODO : we could display the recipe instead of going back to the profile
```
Comment (0marwa) [inline app/src/main/java/com/android/feedme/ui/profile/RecipeInputScreen.kt:118]: Good idea! This seems more logical.

**Review by 0marwa — CHANGES_REQUESTED** [review]
Good job for the new features, this is impressive! (: I don't think there's any major changes to do besides the few comments I've left here and there (documentation and small suggestions). Good idea thinking of redirecting to the `Recipe` page instead of that of the profile after the recipe creation! 
Other than that, the tests have been modified and seem to cover the maximum of lines possible, I can't think of ways to increase coverage as of right now, but we could spend some time on that on another PR, as you suggested. 
Great job overall :)

File: app/src/main/java/com/android/feedme/model/data/ProfileRepository.kt
[the remark this answers is not in the document]
Comment (fdejean) [reply app/src/main/java/com/android/feedme/model/data/ProfileRepository.kt:1]: Hey,
This is a good idea that I wish I could implement but this PR need to be merged before the end of MS4.

File: app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt
[the remark this answers is not in the document]
Comment (fdejean) [reply app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt:1]: ty, done

**Review by 0marwa — APPROVED** [review]
Thank you for addressing the changes requested! This PR can be approved (:

**Pull request merged**